internal/services/containers/kubernetes_cluster_resource.go

```diff
@@ -99,6 +99,18 @@ func resourceKubernetesCluster() *pluginsdk.Resource {
 				},
 			},
 
+			"api_server_vnet_integration_enabled": {
+				Type:     pluginsdk.TypeBool,
+				Optional: true,
+				Default:  false,
+			},
+
+			"api_server_subnet_id": {
+				Type:         pluginsdk.TypeString,
+				Optional:     true,
+				ValidateFunc: azure.ValidateResourceID,
+			},
```

Comment:
Wondering whether it makes sense to move all the `api_server_` properties into a block called `api_server_profile` and deprecate and remove `api_server_authorized_ip_ranges` for 4.0?

Reply:
@stephybun Follow up question: would you advise refactoring all `ManagedClusterAPIServerAccessProfile` properties or only the properties that are named explicitly `api_server_*`?

```go
apiAccessProfile := managedclusters.ManagedClusterAPIServerAccessProfile{
		EnablePrivateCluster:           &enablePrivateCluster,
		AuthorizedIPRanges:             apiServerAuthorizedIPRanges,
		EnablePrivateClusterPublicFQDN: utils.Bool(d.Get("private_cluster_public_fqdn_enabled").(bool)),
		DisableRunCommand:              utils.Bool(!d.Get("run_command_enabled").(bool)),
		EnableVnetIntegration:          &enableAPIServerVnetIntegration,
		SubnetId:                       utils.String(d.Get("api_server_subnet_id").(string)),
	}
```